### PR TITLE
No journal failure for command failure stashed while replaying

### LINKED ISSUE #26256: Stashed bad command during replay causes JournalFailureException
Noted in a test where I expected to be able to send a message that causes a class cast exception, but on travis which is slower than my workstation, it is still replaying when that message is sent, and the message is stashed, to be unstashed when replay completes, however when it then fails it is reported as a JournalFailureException:

```
akka.persistence.typed.internal.JournalFailureException: Exception during recovery.  Last known sequence number [0]. PersistenceId [foiesftr]
	at akka.persistence.typed.internal.ReplayingEvents.onRecoveryFailure(ReplayingEvents.scala:167)
	at akka.persistence.typed.internal.ReplayingEvents.onJournalResponse(ReplayingEvents.scala:108)
	at akka.persistence.typed.internal.ReplayingEvents.$anonfun$stay$1(ReplayingEvents.scala:70)
	at akka.actor.typed.internal.BehaviorImpl$ReceiveMessageBehavior.receive(BehaviorImpl.scala:49)
	at akka.actor.typed.Behavior$.interpret(Behavior.scala:421)
	at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:394)
	at akka.actor.typed.internal.InterceptorImpl$$anon$2.apply(InterceptorImpl.scala:49)
	at akka.actor.typed.internal.WidenedInterceptor.aroundReceive(InterceptorImpl.scala:168)
	at akka.actor.typed.internal.InterceptorImpl.receive(InterceptorImpl.scala:73)
	at akka.actor.typed.Behavior$.interpret(Behavior.scala:421)
	at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:394)
	at akka.actor.typed.internal.InterceptorImpl$$anon$2.apply(InterceptorImpl.scala:49)
	at akka.persistence.typed.internal.EventSourcedBehaviorImpl$$anon$1.aroundReceive(EventSourcedBehaviorImpl.scala:107)
	at akka.actor.typed.internal.InterceptorImpl.receive(InterceptorImpl.scala:73)
	at akka.actor.typed.Behavior$.interpret(Behavior.scala:421)
	at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:394)
	at akka.actor.typed.internal.InterceptorImpl$$anon$2.apply(InterceptorImpl.scala:49)
	at akka.actor.typed.internal.SimpleSupervisor.aroundReceive(Supervision.scala:85)
	at akka.actor.typed.internal.InterceptorImpl.receive(InterceptorImpl.scala:73)
	at akka.actor.typed.Behavior$.interpret(Behavior.scala:421)
	at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:394)
	at akka.actor.typed.internal.adapter.ActorAdapter.akka$actor$typed$internal$adapter$ActorAdapter$$handleMessage(ActorAdapter.scala:82)
	at akka.actor.typed.internal.adapter.ActorAdapter$$anonfun$running$1.applyOrElse(ActorAdapter.scala:78)
	at akka.actor.Actor.aroundReceive(Actor.scala:517)
	at akka.actor.Actor.aroundReceive$(Actor.scala:515)
	at akka.actor.typed.internal.adapter.ActorAdapter.aroundReceive(ActorAdapter.scala:39)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:592)
	at akka.actor.ActorCell.invoke(ActorCell.scala:561)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:258)
	at akka.dispatch.Mailbox.run(Mailbox.scala:225)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:235)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.String
	at akka.persistence.typed.javadsl.Effect.$anonfun$thenRun$1(Effect.scala:102)
	at akka.persistence.typed.javadsl.Effect.$anonfun$thenRun$1$adapted(Effect.scala:102)
	at akka.persistence.typed.internal.Running.applySideEffect(Running.scala:319)
	at akka.persistence.typed.internal.Running.applySideEffects(Running.scala:304)
	at akka.persistence.typed.internal.Running.applyEffects$1(Running.scala:141)
	at akka.persistence.typed.internal.Running.onCommand$1(Running.scala:83)
	at akka.persistence.typed.internal.Running.$anonfun$handlingCommands$3(Running.scala:161)
	at akka.actor.typed.internal.BehaviorImpl$ReceiveMessageBehavior.receive(BehaviorImpl.scala:49)
	at akka.actor.typed.Behavior$.interpret(Behavior.scala:421)
	at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:394)
	at akka.actor.typed.Behavior$.interpretOne$1(Behavior.scala:440)
	at akka.actor.typed.Behavior$.interpretMessages(Behavior.scala:446)
	at akka.actor.typed.internal.StashBufferImpl.unstash(StashBufferImpl.scala:103)
	at akka.persistence.typed.internal.StashManagement.tryUnstash(StashManagement.scala:53)
	at akka.persistence.typed.internal.StashManagement.tryUnstash$(StashManagement.scala:48)
	at akka.persistence.typed.internal.ReplayingEvents.tryUnstash(ReplayingEvents.scala:52)
	at akka.persistence.typed.internal.ReplayingEvents.onRecoveryCompleted(ReplayingEvents.scala:182)
	at akka.persistence.typed.internal.ReplayingEvents.onJournalResponse(ReplayingEvents.scala:98)
	... 33 more
```

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #26256

### CODE REVIEW

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/ReplayingEvents.scala [hunk 1]
```diff
@@ -104,6 +107,7 @@ private[persistence] class ReplayingEvents[C, E, S](override val setup: Behavior
           Behaviors.unhandled
       }
     } catch {
+      case FailureWhileUnstashing(ex) ⇒ throw ex
```
Comment (patriknw) [inline akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/ReplayingEvents.scala]: that looks right, throwing in same way as if the command would be received later in the Running

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by johanandren** [comment]
PLS BUILD

**Comment by patriknw** [comment]
rebase needed

**Comment by patriknw** [comment]
merging so we can start the release

**Pull request merged**